userspace/otpilot/src/sfdp.rs

```diff
@@ -0,0 +1,566 @@
+        //                      A[31:24] bits. Read with instruction C8h. Write
+        //                      instruction is C5h with 1 byte of data. Select the
+        //                      active 128 Mbit memory segment by setting the
+        //                      appropriate A[31:24] bits and use 3-Byte addressing.
+        //           xxxx_1xxxb 8-bit volatile bank register used to define A[30:A24]
+        //                      bits. MSB (bit[7]) is used to enable/disable 4-byte
+        //                      address mode. When MSB is set to ‘1’, 4-byte address
+        //                      mode is active and A[30:24] bits are don’t care. Read
+        //                      with instruction 16h. Write instruction is 17h with 1
+        //                      byte of data. When MSB is cleared to ‘0’, select the
+        //                      active 128 Mbit segment by setting the appropriate
+        //                      A[30:24] bits and use 3-Byte addressing.
+        //           xxx1_xxxxb A 16-bit nonvolatile configuration register controls
+        //                      3-Byte/4-Byte address mode. Read instruction is B5h.
+        //                      Bit[0] controls address mode [0=3-Byte; 1=4-Byte]. Write
+        //                      configuration register instruction is B1h, data length
+        //                      is 2 bytes.
+        //           xx1x_xxxxb Supports dedicated 4-Byte address instruction set.
+        //                      Consult vendor data sheet for the instruction set
+        //                      definition.
+        //           x1xx_xxxxb Always operates in 4-Byte address mode
+        if startup_address_mode == AddressMode::FourByte {
+                1 << 6 // Always operates in 4-Byte address mode
+        } else {
+                if support_address_mode_switch {
+                    1  // issue instruction B7h
+                } else {
+                    0
+                }
+        },
+
+        // Google parameter table
+        0x47, // G
+        0x4f, // O
+        0x4f, // O
+        0x47, // G
+
+        ((mailbox_offset >> 0) & 0xff) as u8,
+        ((mailbox_offset >> 8) & 0xff) as u8,
+        ((mailbox_offset >> 16) & 0xff) as u8,
+        ((mailbox_offset >> 24) & 0xff) as u8,
+
+        ((mailbox_size >> 0) & 0xff) as u8,
+        ((mailbox_size >> 8) & 0xff) as u8,
+        ((mailbox_size >> 16) & 0xff) as u8,
+        ((mailbox_size >> 24) & 0xff) as u8,
+
+        ((google_capabilities >> 0) & 0xff) as u8,
+        ((google_capabilities >> 8) & 0xff) as u8,
+        ((google_capabilities >> 16) & 0xff) as u8,
+        ((google_capabilities >> 24) & 0xff) as u8,
+    ];
+
+    if data.len() < sfdp.len() {
+        return Err(SfdpTableError::TargetLenTooSmall);
+    }
+
+    for idx in 0..sfdp.len() {
+        data[idx] = sfdp[idx];
+    }
+    for idx in sfdp.len()..data.len() {
+        data[idx] = !0;
+    }
+
+    Ok(())
+}
```

Comment:
Nitpick: the last line of this file is missing an EOL character, please add one.

Reply:
One day I'll figure out how to make Sublime do that automatically ... sigh.